src/cim/components/Agent/AgentsSelectionHostCountAlerts.tsx

```diff
@@ -2,37 +2,44 @@ import React from 'react';
 import { Alert, AlertGroup, AlertVariant } from '@patternfly/react-core';
 import ShortCapacitySummary from '../ClusterDeployment/ShortCapacitySummary';
 import { AgentK8sResource } from '../../types';
+import { TFunction } from 'i18next';
 
-const getHostCountWarningText = (hostsSelected: number) => {
+const getHostCountWarningText = (hostsSelected: number, t: TFunction) => {
   switch (hostsSelected) {
     case 0:
-      return 'No host is selected.';
+      return t<string>('ui:No host is selected.');
     case 1:
-      return 'Only 1 host is selected.';
+      return t<string>('ui:Only 1 host is selected.');
   }
-  return `Only ${hostsSelected} hosts are selected.`;
+  return t<string>('ui:Only {{hostsSelected}} hosts are selected.', { hostsSelected });
 };
 
 type AgentsSelectionHostCountAlertsProps = {
   matchingAgentsCount: number;
   selectedAgents: AgentK8sResource[];
   targetHostCount: number;
+  t: TFunction;
```

Comment:
we should not pass `t` as prop to components.
Just use hook to get one

Reply:
i am using it in the same way that in the console project: https://github.com/stolostron/console/blob/main/frontend/src/lib/validation.ts#L9